### PR TITLE
chore: transfer utils's interval map impl into dependency

### PR DESCRIPTION
**Pull request by GG2002**
Please briefly answer these questions:

* what problem are you trying to solve? (or if there's no problem, what's the motivation for this change?)

* what changes does this pull request make?
  Transfer `utils`'s interval map implementation into dependency on `rb-interval-map` which is based on `utils`'s implementation.
* are there any non-obvious implications of these changes? (does it break compatibility with previous versions, etc)
  No.

### CODE REVIEW

File: crates/utils/src/lib.rs [hunk 1]
```diff
@@ -211,6 +209,8 @@ pub mod tokio_lock;
 pub mod tracing;
 
 use ::tracing::debug;
+/// Interval tree implementation
+pub use interval_map;
```
Comment (bsbds) [inline crates/utils/src/lib.rs:213]: We could directly use interval_map in xline crate.

Comment (GG2002) [reply crates/utils/src/lib.rs:213]: That would require changing the dependencies of two crates, `xline` and `xlineapi`. Is that okay? Now all we need to do is modify the `utils`.

**Review by bsbds — APPROVED** [review]

File: crates/utils/Cargo.toml [hunk 2]
```diff
@@ -25,12 +25,11 @@ derive_builder = "0.20.0"
 event-listener = "5.3.1"
 futures = "0.3.30"
 getset = "0.1"
+interval_map = { package = "rb-interval-map" }
```
Comment (lxl66566) [inline crates/utils/Cargo.toml:28]: how about adding `version = "0.1.2"`? This line caused an error on rust-analyzer:
```
2024-08-26T00:21:28.930391Z ERROR FetchWorkspaceError
...
Caused by:
  dependency (interval_map) specified without providing a local path, Git repository, version, or workspace dependency to use
```

Comment (GG2002) [reply crates/utils/Cargo.toml:28]: adding `version = "0.1"` is ok, as this crate might require further modifications

**Pull request merged**